httpclient5/src/main/java/org/apache/hc/client5/http/impl/nio/PoolingAsyncClientConnectionManager.java

```diff
@@ -252,7 +252,7 @@ public void completed(final PoolEntry<HttpRoute, ManagedAsyncClientConnection> p
 
                                     @Override
                                     public void execute(final Boolean result) {
-                                        if (result == Boolean.FALSE) {
+                                        if (!result) {
```

Comment:
Can this actually lead to an NPE?

Reply:
I had initially written it this way to retain identical semantics:

```suggestion
                                        if (result != null && !result) {
```

But it's unclear what null would mean if it's possible, and further analysis indicated that nothing passed null values.